src/FlatButton/FlatButton.js

```diff
@@ -6,7 +6,7 @@ import EnhancedButton from '../internal/EnhancedButton';
 import FlatButtonLabel from './FlatButtonLabel';
 
 function validateLabel(props, propName, componentName) {
-  if (!props.children && !props.label && !props.icon) {
+  if (!props.children && (!props.label && props.label !== 0) && !props.icon) {
```

Comment:
@oliviertassinari 
- I don't think any warning is needed here. I am wondering to know, warning message for what ? Component user has not done any mistake if he passed 0 as label.
- Actually, Label validation doesn't depends on single property. We need to check presence of anyone property out of children, label, and icon. So we can't.